### DISCUSSION TITLE
(berry) how do iteration over all objects/instaces from a class?

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#18179] [Q&A]
I would like to access all instances (buttons) of a class "Button" with a static method. For example, I want to query the GPIO(input) status of all instances after a trigger. Are there any built-in functions for this?

My workaround is now to initialize all instances in a list and perform the iteration with a function outside of the class.

**Comment by sfromis** [comment]
You could have your class record it's instance creations in a global variable (like a list). That would also be available to a static method in the class.
